### PR TITLE
fix(ui): Various react 18 type changes

### PR DESCRIPTION
**Pull request by scttcper**
There's a few places where we're passing objects to ReactNode, most of them can be casted. Added a few TODOS.

got tired of making a million little pull requests and these feel similar.

part of https://github.com/getsentry/frontend-tsc/issues/22

### CODE REVIEW

File: static/app/views/projectDetail/projectLatestAlerts.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ function AlertRow({alert, orgSlug}: AlertRowProps) {
       aria-label={title}
       to={`/organizations/${orgSlug}/alerts/${identifier}/`}
     >
-      <AlertBadgeWrapper {...statusProps} icon={Icon}>
+      <AlertBadgeWrapper {...statusProps} icon={<Icon />}>
```
Comment (scttcper) [inline static/app/views/projectDetail/projectLatestAlerts.tsx:45]: I think this is the only type change. Its declared a few lines up and is a component

File: static/app/components/gridEditable/index.tsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ type GridEditableProps<DataRow, ColumnKey> = {
   };
   location: Location;
   emptyMessage?: React.ReactNode;
-  error?: React.ReactNode | null;
```
Comment (scttcper) [inline static/app/components/gridEditable/index.tsx:98]: in this case it was never a ReactNode its an Error object of some kind. We just check if it exists so unknown is okay to check truthy

File: static/app/types/group.tsx [hunk 3]
```diff
@@ -391,7 +391,7 @@ export interface GroupActivityNote extends GroupActivityBase {
 }
 
 interface GroupActivitySetResolved extends GroupActivityBase {
-  data: Record<string, unknown>;
+  data: {};
```
Comment (scttcper) [inline static/app/types/group.tsx:394]: it didn't like that we might be rendering unknown to the dom, this removes the keys

File: static/app/components/dropdownMenu/item.tsx [hunk 4]
```diff
@@ -215,7 +215,7 @@ function BaseDropdownMenuItem(
       trailingItems={
         isSubmenu ? (
           <Fragment>
-            {trailingItems}
+            {trailingItems as React.ReactNode}
```
[the remark this answers is not in the document]
Comment (scttcper) [reply static/app/components/dropdownMenu/item.tsx:218]: it appears it is sometimes https://github.com/getsentry/sentry/blob/43cac31867cd035e2ad6f0c4108cc8bb60726082/static/app/components/menuListItem.tsx#L155-L157

**Review by malwilley — APPROVED** [review]

**Pull request merged**